### PR TITLE
Add BlueStyleBadge and ColPracBadge Plugins

### LINKED ISSUE #227: How to add custom badge?
I'd like to add the ability to generate a repo with both
- a [BlueStyle badge](https://github.com/invenia/BlueStyle#code-style-badge)
- a [ColPrac badge](https://github.com/SciML/ColPrac#marking-a-repository-as-following-colprac)

AFAICT, badges are added to the README via `badges(::Plugin)`, but it's not clear how to have additional badges not connected to a `Plugin`

Does one have to implement a `Plugin`? e.g.
```julia
struct BlueStyleBadge <: Plugin end
badges(::BlueStyleBadge) = Badge(
    "Code Style: Blue",
    "https://img.shields.io/badge/code%20style-blue-4495d1.svg",
    "https://github.com/invenia/BlueStyle"
)
```

### PR DESCRIPTION
**Pull request by nickrobinson251**
- closes #227 
- `BadgePlugin` created to avoid need to update `Readme` Plugin with future badges

Questions:
- is this code in a sensible place?
- are the tests sensible and sufficient?
- are the tests in a sensible place?

### CODE REVIEW

File: src/plugins/badges.jl [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+Adds a [`ColPrac`](https://github.com/SciML/ColPrac) badge to the [`Readme`](@ref) file.
+"""
+struct ColPracBadge <: BadgePlugin end
+
+function badges(::ColPracBadge)
+    return Badge(
+        "ColPrac: Contributor's Guide on Collaborative Practices for Community Packages",
+        "https://img.shields.io/badge/ColPrac-Contributor's%20Guide-blueviolet",
+        "https://github.com/SciML/ColPrac"
```
Comment (christopher-dG) [inline src/plugins/badges.jl:32]:
```suggestion
        "https://github.com/SciML/ColPrac",
```

File: src/plugins/badges.jl [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+Adds a [`BlueStyle`](https://github.com/invenia/BlueStyle) badge to the [`Readme`](@ref) file.
+"""
+struct BlueStyleBadge <: BadgePlugin end
+
+function badges(::BlueStyleBadge)
+    return Badge(
+        "Code Style: Blue",
+        "https://img.shields.io/badge/code%20style-blue-4495d1.svg",
+        "https://github.com/invenia/BlueStyle"
```
Comment (christopher-dG) [inline src/plugins/badges.jl:17]:
```suggestion
        "https://github.com/invenia/BlueStyle",
```

**Comment by nickrobinson251** [comment]
oh looks like these need to be added to the docs manually? i'll do that

**Comment by nickrobinson251** [comment]
Apparently I'm "not not authorized to merge this pull request."

But it should be good to go, if someone can merge it for me or give me permissions?

**Pull request merged**